### ISSUE TITLE
Support STATEMENT_TYPE negation for 'padding-line-between-statements'

### ISSUE THREAD

**Issue by roydukkey** [eslint/eslint#13800]
**What rule do you want to change?**
`padding-line-between-statements`

**Does this change cause the rule to produce more or fewer warnings?**
N/A

**How will the change be implemented? (New option, new default behavior, etc.)?**
Should allow `STATEMENT_TYPE` to be negated. Possibly with a bang (!).

**Please provide some example code that this change will affect:**

```json
"padding-line-between-statements": [
	"error",
	{
		"blankLine": "always",
		"prev": "import",
		"next": "*"
	}
]
```

```js
import Scope from './Scope';
import { join } from 'path';
import { mkdir, writeFile } from 'fs/promises';

class CompilationOptions {
}
```

**What does the rule currently do for this code?**
Shows the error for line 2-3:

```js
import { join } from 'path'; // Expected blank line before this statement. eslint(padding-line-between-statements)
import { mkdir, writeFile } from 'fs/promises'; // Expected blank line before this statement. eslint(padding-line-between-statements)
```

**What will the rule do after it's changed?**

```json
"padding-line-between-statements": [
	"error",
	{
		"blankLine": "always",
		"prev": "import",
		"next": ["*", "!import"]
	}
]
```

```js
import Scope from './Scope';
import { join } from 'path';
import { mkdir, writeFile } from 'fs/promises';

class CompilationOptions {
}
```

**Are you willing to submit a pull request to implement this change?**
No. Sorry.

**Comment by mdjermanovic** [comment]
Hi @roydukkey, thanks for the issue!

This is already supported by how the configuration for this rule works. For a pair of statement, it will apply the _last_ match in the config, so you can configure the rule to ignore pairs of imports while still requiring a blank line after an import if it's followed by any other statement, like this:

```json
"padding-line-between-statements": [
	"error",
	{
		"blankLine": "always",
		"prev": "import",
		"next": "*"
	},
        {
		"blankLine": "any",
		"prev": "import",
		"next": "import"
	}
]
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwicGFkZGluZy1saW5lLWJldHdlZW4tc3RhdGVtZW50c1wiOiBbXG5cdFwiZXJyb3JcIixcblx0e1xuXHRcdFwiYmxhbmtMaW5lXCI6IFwiYWx3YXlzXCIsXG5cdFx0XCJwcmV2XCI6IFwiaW1wb3J0XCIsXG5cdFx0XCJuZXh0XCI6IFwiKlwiXG5cdH0sXG4gICAge1xuXHRcdFwiYmxhbmtMaW5lXCI6IFwiYW55XCIsXG5cdFx0XCJwcmV2XCI6IFwiaW1wb3J0XCIsXG5cdFx0XCJuZXh0XCI6IFwiaW1wb3J0XCJcblx0fVxuXSovXG5cbmltcG9ydCBTY29wZSBmcm9tICcuL1Njb3BlJztcbmltcG9ydCB7IGpvaW4gfSBmcm9tICdwYXRoJztcbmltcG9ydCB7IG1rZGlyLCB3cml0ZUZpbGUgfSBmcm9tICdmcy9wcm9taXNlcyc7XG5cbmNsYXNzIENvbXBpbGF0aW9uT3B0aW9ucyB7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**Comment by roydukkey** [comment]
I see thanks!
